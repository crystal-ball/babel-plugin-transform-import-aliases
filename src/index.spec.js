'use strict'

const babel = require('@babel/core')

const plugin = require('./index')

// Runs relative to repo root
const content = `import Rad from '@/dux/rad'
import { css } from '@emotion/core'
import { ENV } from 'ENV/dev'
const HomeScreen = import(/* webpackChunkName: "HomeScreen" */ '@/screens/HomeScreen/HomeScreen')

function load() {}
const FakeScreen = load(/* webpackChunkName: "HomeScreen" */ '@/screens/HomeScreen/HomeScreen')
`

it('transforms imports', async () => {
  const { code } = await babel.transformAsync(content, {
    plugins: [[plugin, { aliases: { '@': '/User/Rad/Code/src', 'ENV': '/tmp/env' } }]],
    // Tests the relative alias transforms
    filename: '/User/Rad/Code/src/components/App/App.ts',
  })

  expect(code).toMatchSnapshot()
})
