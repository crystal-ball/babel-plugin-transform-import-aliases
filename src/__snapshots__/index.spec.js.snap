// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms imports 1`] = `
"import Rad from \\"../../dux/rad\\";
import { css } from '@emotion/core';
import { ENV } from \\"/tmp/env/dev\\";
const HomeScreen = import(
/* webpackChunkName: \\"HomeScreen\\" */
\\"../../screens/HomeScreen/HomeScreen\\");

function load() {}

const FakeScreen = load(
/* webpackChunkName: \\"HomeScreen\\" */
'@/screens/HomeScreen/HomeScreen');"
`;
